public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -438,8 +438,13 @@ export class LokiDatasource
   }
 
   async metadataRequest(url: string, params?: Record<string, string | number>) {
-    const res = await lastValueFrom(this._request(url, params, { hideFromInspector: true }));
-    return res.data.data || res.data.values || [];
+    if (config.featureToggles.lokiBackendMode) {
+      const res = await this.getResource(url, params);
+      return res.data || res.values || [];
+    } else {
+      const res = await lastValueFrom(this._request(url, params, { hideFromInspector: true }));
+      return res.data.data || res.data.values || [];
```

Comment:
I guess, Ww can also remove it form here.

Reply:
i agree. but the situation is more complicated than i thought, so i recommend handling these in a separate PR.